configure/os/CONFIG.Common.RTEMS

```diff
@@ -102,15 +102,38 @@ OS_CLASS = RTEMS
 # Operating system compile & link flags
 OP_SYS_CFLAGS += -D__LINUX_ERRNO_EXTENSIONS__
 
-OP_SYS_CFLAGS_NET_yes = -DRTEMS_LEGACY_STACK
-OP_SYS_CFLAGS += $(OP_SYS_CFLAGS_NET_$(RTEMS_HAS_NETWORKING))
+# Has RTEMS been built with the legacy stack as a separate package?
+ifeq ($(RTEMS_LEGACY_NETWORKING),yes)
+RTEMS_HAS_NETWORKING = yes
+RTEMS_NETWORKING = legacy
+endif
+
+# Has RTEMS been built with the libbsd stack as a separate package?
+ifeq ($(RTEMS_BSD_NETWORKING),yes)
+RTEMS_HAS_NETWORKING = yes
+RTEMS_NETWORKING = bsd
+endif
+
+RTEMS_LEGACY_NET_LIB_no=
+
+# Leagcy network with RTEMS 6 is a separate package and library
+RTEMS_NET_LIB_legacy=-lnfs -lnetworking -lnfs
+OP_SYS_CFLAGS_NET_legacy = -DRTEMS_LEGACY_STACK
+
+# LibBSD network with RTEMS 5 and 6 is a separate package and library
+RTEMS_NET_LIB_bsd=-lbsd
+OP_SYS_CFLAGS_NET_bsd = -DRTEMS_LIBBSD_STACK
+
+# Set the networking flags
+OP_SYS_CFLAGS += $(OP_SYS_CFLAGS_NET_$(RTEMS_NETWORKING))
 
 ifeq ($(RTEMS_HAS_POSIX_API),yes)
 POSIX_CPPFLAGS = -D_GNU_SOURCE -D_DEFAULT_SOURCE
 endif
 
-OP_SYS_LDLIBS_posix_NET_yes = -ltftpfs -lnfs -lz -ltelnetd
-OP_SYS_LDLIBS_posix_NET_no = -ltftpfs -lbsd -lz
+OP_SYS_LDLIBS_posix_NET_yes = -ltftpfs -lz -ltelnetd -lntp
```

Comment:
`libntp` is provided by `rtems-net-services` repo. Does this change make `rtems-net-services` a requirement to build with any networking stack?

Reply:
I seems to have pulled down the `libntp` library change from the NTP patch. I can rework the patch set if this is a problem? The RSB can build tools, kernel, networking and net services as a build set for suitable BSPs such as the MVME2700.

> Does this change make `rtems-net-services` a requirement to build with any networking stack?

Good question and I am not in a position to answer this. It depends on what the core team see as the base requirements for an RTEMS EPICS build. My input is to say "yes" and my reasons are...

EPICS 3 + RTEMS 4.10 provided NTP via the EPICS built in client so this functionality is expected. RTEMS 6 + EPICS 7 will support NTP via `ntpd` and not the EPICS NTP client. The `ntpd` solution provides functional backward compatibility at the NTP level with the improved performance and feature set `ntpd` has.

My analysis of the history is changes by Heinz for RTEMS on EPICS 7 commented out the EPICS NTP client support in RTEMS specific code as `libbsd` was the only networking option and it does not have the code the EPICS NTP client depends on. This situation has changed with the legacy networking stack now available however I would encourage the use of `ntpd` on RTEMS over the EPICS NTP support.

The EPICS NTP client for RTEMS depended on non-standard calls to the handle the NTP response that only exist in the legacy code. I do not see that code being moved anywhere else.

The topic of `ntpd` is not the focus of this PR and I suggest it be discussed in a new issue or a PR for that change. I have a branch in my repo that adds support. We are close to finishing our testing at Gemini and my recent changes look good.

So back to your original question. I feel building net services and having `libntpd` will happen if you want an NTP solution that matches other systems like Linux, Window, MacOS etc.